lib/maths/CBoostedTreeLeafNodeStatistics.cc

```diff
@@ -398,30 +479,47 @@ CBoostedTreeLeafNodeStatistics::computeBestSplitStatistics(const TRegularization
             hr[ASSIGN_MISSING_TO_LEFT] -= curvature;
             hr[ASSIGN_MISSING_TO_RIGHT] -= curvature;
 
-            double gain[2];
-            gain[ASSIGN_MISSING_TO_LEFT] =
-                cl[ASSIGN_MISSING_TO_LEFT] == 0 || cl[ASSIGN_MISSING_TO_LEFT] == c
-                    ? -INF
-                    : minimumLoss(gl[ASSIGN_MISSING_TO_LEFT], hl[ASSIGN_MISSING_TO_LEFT]) +
-                          minimumLoss(gr[ASSIGN_MISSING_TO_LEFT], hr[ASSIGN_MISSING_TO_LEFT]);
-            gain[ASSIGN_MISSING_TO_RIGHT] =
-                cl[ASSIGN_MISSING_TO_RIGHT] == 0 || cl[ASSIGN_MISSING_TO_RIGHT] == c
-                    ? -INF
-                    : minimumLoss(gl[ASSIGN_MISSING_TO_RIGHT], hl[ASSIGN_MISSING_TO_RIGHT]) +
-                          minimumLoss(gr[ASSIGN_MISSING_TO_RIGHT],
-                                      hr[ASSIGN_MISSING_TO_RIGHT]);
+            if (cl[ASSIGN_MISSING_TO_LEFT] == 0 || cl[ASSIGN_MISSING_TO_LEFT] == c) {
+                gain[ASSIGN_MISSING_TO_LEFT] = -INF;
+            } else {
+                minLossLeft[ASSIGN_MISSING_TO_LEFT] = minimumLoss(
```

Comment:
Does it matter that we only set `minLossLeft` in this branch?  I don't think so because they are only used if `gain > -INF`, but this is not immediately obvious from the code. I think it warrants a comment.

Reply:
You are right, I added a comment.